cloudinit/sources/DataSourceVMware.py

```diff
@@ -776,91 +774,52 @@ def get_default_ip_addrs():
     addresses associated with the device used by the default route for a given
     address.
     """
-    # TODO(promote and use netifaces in cloudinit.net* modules)
-    gateways = netifaces.gateways()
-    if "default" not in gateways:
-        return None, None
-
-    default_gw = gateways["default"]
-    if (
-        netifaces.AF_INET not in default_gw
-        and netifaces.AF_INET6 not in default_gw
-    ):
-        return None, None
-
     ipv4 = None
     ipv6 = None
+    routes = netinfo.route_info()
+    for route in routes:
```

Comment:
I'm not familiar with netinfo.route_info(), but with my local check,  it returns a dict but list, so is this loop working as expected?

Reply:
netinfo.route_info() has _netdev_route_info_iproute and _netdev_route_info_netstat as the underlying functions, both of which should return a list of dicts with keys "destination", "gateway", "metric", etc.  What circumstances are you seeing something different?